File: cpp/prob4/prob4.cpp
#include <iostream> 
#include <fstream> 

using namespace std;


int main() {
    ifstream myfile ("OliverProb4Input.txt");
    string line; 
    int nums[4]; 
    int numsLength = 4; 
    int total;
    bool print = true;
    string ch;

    while(getline(myfile, line)) {
        for (int i = 0; i < numsLength; i++) {
            for(int j = 0; j < line.size(); j++) {
                ch += line[j];
                if(ch == "-" || ch == ",") break; 
                nums[i] *= 10; 
                nums[i] += stoi(ch);
                ch = "";
            }
        }
        if(print) {
            for(int i : nums) {
                cout << i << " ";
            }
            cout << endl; 
            print = false;
        }
        
        if(nums[0] >= nums[2] && nums[1] <= nums[3]) total++;
        if(nums[0] < nums[2] && nums[1] > nums[3]) total++;
    }
    cout << total << endl;
    return 0;
}

File: python/prob4/prob4.py
nums = []
total = 0
with open('OliverProb4Input.txt') as file:
    total = 0
    for line in file: 
        numIdx = 0
        nums = []
        for item in line.split(","):
            for num in item.split("-"):
                nums.append(int(num.strip()))
        if(nums[0] <= nums[2] and nums[1] >= nums[3]):
            total += 1
        elif(nums[2] <= nums[0] and nums[3] >= nums[1]): 
            total += 1
        
        
print(total)